Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2557 -- Show clickable subject tags in Combined Search page results

Resolves [EREGCSC-2557](https://jiraent.cms.gov/browse/EREGCSC-2557)

**Description**

Let's display subjects in the resources search results on the combined search page.

Clicking a subject tag would take you to that subject page (no other parameters).

**This pull request changes:**

- Shows subject tags on the Search page
   - They are styled exactly the same as the subject tags on the Subjects page
   - They are clickable in the same way they are on the Subjects page: clicking a subject tag will send the user to the Subjects page with the clicked subject selected as a filter
- Extends `SubjectChip` to use anchor tag for clickable Subject tag when used on pages that are not the Subjects page
   - Follows the `isExternalLink` pattern used in first example here: https://router.vuejs.org/guide/advanced/extending-router-link.html
   - Adds e2e test

**Steps to manually verify this change:**

1. Visit [experimental deployment](https://l9cen2wicj.execute-api.us-east-1.amazonaws.com/dev1204/) and log in
2. Visit the search page and search for docs that have attached subjects
   - Example: [faraway](https://l9cen2wicj.execute-api.us-east-1.amazonaws.com/dev1204/search/?q=faraway)
3. Click one of the subject chips
   - Example: "Autism Spectrum Disorder"
5. Ensure you arrives at `/subjects` with the appropriate subject selected
   - Example: [Autism Spectrum Disorder](https://l9cen2wicj.execute-api.us-east-1.amazonaws.com/dev1204/subjects/?subjects=7)
